### ISSUE TITLE
Implement updates to the nullifier DB on `apply_block`

### ISSUE THREAD

**Issue by bobbinth** [0xPolygonMiden/miden-node#140]
Currently, we do not update nullifier database on apply block. To fix this, we should do the following:

1. Make sure store's `GetBlockInputs` endpoint returns all relevant data needed by the block producer to update the nullifier DB root.
2. Update the block producer (and the block kernel) to compute the new nullifier DB root.
3. Update the store's `apply_block` handler to make relevant updates to the nullifier DB (this may be already done).

Before we can do the above, we need to complete implementing the alternative to TSMT in miden crypto/vm.

**Comment by bobbinth** [comment]
@plafer - this should be closed now too, right?

**Comment by plafer** [comment]
Closed by #209
